zerver/tests/test_archive.py

```diff
@@ -9,7 +9,7 @@ def test_non_existant_stream_id(self) -> None:
         # Here we use a relatively big number as stream id assumming such an id
         # won't exist in the test DB.
         result = self.client_get("/archive/streams/100000000/topic/TopicGlobal")
-        self.assert_json_error(result, 'Invalid stream id')
+        self.assert_in_success_response(["This stream does not exist."], result)
```

Comment:
Once we merge the main infra commits, I think we can split off a project to make a better error page for this case; I think we'll want to show a list of the public streams in the organization.